File: app/src/main/java/com/example/dev/hazikura/fragment/Household/DatePick2.java
package com.example.dev.hazikura.fragment.Household;

import android.app.DatePickerDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.widget.DatePicker;
import android.widget.TextView;

import com.example.dev.hazikura.R;

import java.util.Calendar;
import java.util.Locale;

public class DatePick2 extends DialogFragment implements DatePickerDialog.OnDateSetListener {

    @Override
    @NonNull
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Calendar calendar = Calendar.getInstance();
        int year = calendar.get(Calendar.YEAR);
        int month = calendar.get(Calendar.MONTH);
        int dayOfMonth = calendar.get(Calendar.DAY_OF_MONTH);

        DatePickerDialog datePickerDialog = new DatePickerDialog(getActivity(),this, year, month, dayOfMonth);

        return datePickerDialog;
    }

    public void onDateSet(DatePicker view, int y, int m, int d) {
        final TextView textView = getActivity().findViewById(R.id.display_date2);
        String str = String.format(Locale.US, "%d/%d/%d", y, m+1, d);
        textView.setText(str);
    }

}

File: app/src/main/java/com/example/dev/hazikura/fragment/household/DatePick2.kt
package com.example.dev.hazikura.fragment.household

import android.app.DatePickerDialog
import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.widget.DatePicker
import android.widget.TextView

import com.example.dev.hazikura.R

import java.util.Calendar
import java.util.Locale

/**
 * Created by yokoro
 */

class DatePick2 : DialogFragment(), DatePickerDialog.OnDateSetListener {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val calendar = Calendar.getInstance()
        val year = calendar.get(Calendar.YEAR)
        val month = calendar.get(Calendar.MONTH)
        val dayOfMonth = calendar.get(Calendar.DAY_OF_MONTH)

        return DatePickerDialog(activity, this, year, month, dayOfMonth)
    }

    override fun onDateSet(view: DatePicker, y: Int, m: Int, d: Int) {
        val textView = activity?.findViewById<TextView>(R.id.output_date)
        val str = String.format(Locale.US, "%d/%d/%d", y, m + 1, d)
        textView?.text = str
    }

}